mito-ai/src/Extensions/AiChat/ChatMessage/ChatDropdown.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { ExpandedVariable } from './ChatInput';
+
+interface ChatDropdownProps {
+    options: ExpandedVariable[];
+    selectedIndex: number;
+    onSelect: (variableName: string, parentDf?: string) => void;
+}
+
+const ChatDropdown: React.FC<ChatDropdownProps> = ({
+    options,
+    selectedIndex,
+    onSelect,
+}) => {
+    const getShortType = (type: string) => {
+        return type.includes("DataFrame") ? "df"
+            : type.includes("<class '") ? type.split("'")[1]
+                : type;
+    }
+
+    return (
+        <ul className="chat-dropdown-list">
```

Comment:
I think there's two improvements we should make here: 

1. Instead of showing every option by default, maybe we should only show the first 5 or so. Having this giant list gets a little bit out of hand. 
2. We should position the dropdown above the taskpane using the `z-index`. Just like when you right click to inspect your browser, the dropdown appears above the browser instead of as part of the webpage. This is the same with @ mentions in Slack, Zulip, Cursor, etc.

Reply:
For example, right now, I can scroll down the taskpane to see this entire dropdown and not even see my chat input anymore. 

<img width="454" alt="Screenshot 2024-11-14 at 7 24 47 PM" src="https://github.com/user-attachments/assets/f221bb36-550a-4298-832e-64559b99c4ab">

Reply:
Another example, when I've been using the chat taskpane for a bit and therefore the chat is full, the @ mention dropdown is opened off the bottom of the visible taskpane so I didn't even notice it right away. This is another reason why having the dropdown hover will be nice because it can open up instead of down, like cursor (see below)
<img width="380" alt="Screenshot 2024-11-14 at 7 27 41 PM" src="https://github.com/user-attachments/assets/945ccdd2-31ea-4514-8f3f-b1d43a69a364">

<img width="446" alt="Screenshot 2024-11-14 at 7 31 30 PM" src="https://github.com/user-attachments/assets/8e783670-f5cb-4979-8dfb-77b0be55535f">

Reply:
> We should position the dropdown above the taskpane using the z-index.

Might need some more clarity on this. `.chat-dropdown-list` has a z-index of 9999, which should ensure it's always on top.